src/main/java/com/redhat/cloud/notifications/GwResource.java

```diff
@@ -125,17 +134,32 @@ public Response forward(@NotNull @Valid RestAction ra) {
 
         Action action = builder.build();
         String serializedAction = Parser.encode(action);
-        emitter.send(buildMessageWithId(serializedAction));
-        forwardedActions.increment();
 
-        return Response.ok().build();
+        CompletableFuture<Void> callback = new CompletableFuture<>();
+        emitter.send(buildMessageWithId(serializedAction, callback));
+        try {
+            callback.get(callbackTimeout, TimeUnit.SECONDS);
+            forwardedActions.increment();
+            return Response.ok().build();
+        } catch (Exception e) {
+            Log.error("Message delivery to Kafka failed", e);
+            return Response.status(SERVICE_UNAVAILABLE).build();
```

Comment:
@jharting Does returning 503 look good to you if we failed to deliver the Kafka message?